modules/local/Flomics_QC_agreggator.nf

```diff
@@ -102,14 +102,8 @@ process FLOMICS_QC_AGGREGATOR{
         fi
     done < samples.tsv
 
-    echo -e "Sample\tnumber_of_genes_contributing_to_1%_of_reads\tnumber_of_genes_contributing_to_5%_of_reads\tnumber_of_genes_contributing_to_10%_of_reads\tnumber_of_genes_contributing_to_50%_of_reads\tnumber_of_genes_contributing_to_80%_of_reads" > library_balance.tsv
-    while IFS= read -r sample; do
-        if test -f "${sample}_genes_contributing_to_percentage_reads.tsv"; then
-            tail -n +2 ${sample}_genes_contributing_to_percentage_reads.tsv| sed "s/^/$sample\t/" >> library_balance.tsv
-        else
-            echo -e "$sample\tNA\tNA\tNA\tNA\tNA" >> library_balance.tsv
-        fi
-    done < samples.tsv
+    awk '{print $1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6}' genes_contributing_to_percentage_reads.tsv | head -n 1 > library_balance.tsv
+    awk '{print $1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6}' genes_contributing_to_percentage_reads.tsv | tail -n+2 | sort -k1,1 >> library_balance.tsv
```

Comment:
This will be improved a lot when migrating the script to python (aligning/sorting columns based on column names and aligning indices when joining tables, etc). In the current state, this is difficult to maintain, for example, if we add one column to the table (nb of genes contributing to 95% of reads), several lines of codes have to be modified in different places.

Reply:
@ctuni please compare aggregated results with input of this process